File: benchmark/comparisons/script/scripts/table_lookup.py
# table_lookup.py -- N inserts of (i -> i*2) into a dict, then N sequential
# lookups summing the values. Lookup loop in a function so loop vars are locals.

import sys
import time


def build_dict(n):
    d = {}
    for i in range(n):
        d[i] = i * 2
    return d


def sum_lookup(d, n):
    total = 0
    for i in range(n):
        total += d[i]
    return total


count = int(sys.argv[1])
d = build_dict(count)

t0 = time.perf_counter_ns()
total = sum_lookup(d, count)
elapsed = time.perf_counter_ns() - t0
print(elapsed)

File: benchmark/comparisons/script/scripts/table_lookup.lua
-- table_lookup.lua -- N inserts of (i -> i*2) into a table, then N
-- sequential lookups summing the values. Only the lookup loop is timed.

local count = tonumber(arg[1])
local d = {}
for i = 0, count - 1 do
    d[i] = i * 2
end

local t0 = os.clock()

local total = 0
for i = 0, count - 1 do
    total = total + d[i]
end

local elapsed_ns = (os.clock() - t0) * 1e9
print(math.floor(elapsed_ns))
